packages/@aws-cdk/core-synthesizer/lib/synthesizer.ts

```diff
@@ -0,0 +1,79 @@
+import { App, AssetManifestBuilder, DockerImageAssetLocation, DockerImageAssetSource, FileAssetLocation, FileAssetSource, IBoundStackSynthesizer, IReusableStackSynthesizer, ISynthesisSession, Stack, StackSynthesizer } from '@aws-cdk/core';
+import { assertBound } from '@aws-cdk/core/lib/stack-synthesizers/_shared';
+import { IStagingStack, StagingStack } from './staging-stack';
+
+/**
+ * New stack synthesizer properties
+ */
+export interface NewStackSynthesizerProps {
+  /**
+   * Bring a custom staging stack into the app.
+   *
+   * @default - default staging stack
```

Comment:
> A new staging stack is created for every environment that stacks are deployed to

?

Reply:
having a bit of trouble thinking through how to get app-level `StagingStack` passed through the stack synthesizers, since they seem to be operating on different scope. I think that means that I need to keep track of what staging stacks have been created as we are recursing through the tree from the root and finding stacks to synthesize (in `core/lib/private/synthesis.ts`).